lib/Pool.js

```diff
@@ -317,6 +344,10 @@ class Pool {
         return new Promise(async (resolve, reject) => {
           try {
             await this.increasePoolSize(this.initialSize);
+            // Try to get one connection during init to make sure there were
+            // no errors with the connection string
+            const connection = await this.connect();
+            connection.close();
```

Comment:
Is this connection being pulled from the pool and released back or are we making a single connection which is freed just to check for errors?

Reply:
This is a good point. Currently it just called `close` which will either completely close the connection *OR* return it to the pool depending on what the user has specified for `reuseConnections`. I should just `unshift` the connection back onto `freeConnections` after connect hasn't thrown.